core/src/com/unciv/logic/civilization/TechManager.kt

```diff
@@ -334,7 +318,23 @@ class TechManager : IsPartOfGameInfoSerialization {
             civInfo.addNotification("You have unlocked [The Long Count]!", MayaLongCountAction(), MayaCalendar.notificationIcon)
         }
 
+        val previousEra = civInfo.getEra()
         updateEra()
+        val currentEra = civInfo.getEra()
+        if (previousEra != currentEra) {
+            civInfo.addNotification("You have entered the [$currentEra]!", NotificationIcon.Science)
+            if (civInfo.isMajorCiv()) {
+                for (knownCiv in civInfo.getKnownCivs()) {
+                    knownCiv.addNotification("[${civInfo.civName}] has entered the [$currentEra]!", civInfo.civName, NotificationIcon.Science)
```

Comment:
Looking at this makes me realize that it might be worth adding an "Extra Notifications" option to the game settings where players can opt to get a lot more notifications with extra information like this. I think a more avid Unciv player in a multiplayer game might want to know when this happens but for a casual player this is probably irrelevant (other than for effects that happen once a player does something like advancing to an era where all civs gain a spy since that obviously has an effect on the player).

Reply:
In this case I've just moved existing code to a different spot so that it actually works again. Additionally this is a notification that is shown in the original, even for unknown civs. It also doesn't happen too often anyway.
In general that might be a good idea though